gapis/api/vulkan/overdraw.go

```diff
@@ -2204,3 +2207,941 @@ func (*stencilOverdraw) Flush(ctx context.Context, out transform.Writer) {}
+				VkDependencyFlags(0),
+				0,
+				memory.Nullptr,
+				0,
+				memory.Nullptr,
+				1,
+				resolveImageToSrcBarrierData.Ptr(),
+			).AddRead(
+				resolveImageToSrcBarrierData.Data(),
+			),
+		)
+	}
+
+	// Blit image, if the attachment image is multi-sampled, use resolve image as the source image, otherwise
+	// use the attachment image as the source image directly
+	blitSrcImage := imageObject.VulkanHandle()
+	if imageObject.Info().Samples() != VkSampleCountFlagBits_VK_SAMPLE_COUNT_1_BIT {
+		blitSrcImage = resolveImageID
+	}
+	// If the src image is a depth/stencil image, the filter must be NEAREST
+	filter := VkFilter_VK_FILTER_LINEAR
+	if aspect != VkImageAspectFlagBits_VK_IMAGE_ASPECT_COLOR_BIT {
+		filter = VkFilter_VK_FILTER_NEAREST
+	}
+
+	copySrc := blitSrcImage
+
+	if doBlit {
+		copySrc = stagingImageID
+		writeEach(ctx, out,
+			cb.VkCmdBlitImage(
+				commandBufferID,
+				blitSrcImage,
+				VkImageLayout_VK_IMAGE_LAYOUT_TRANSFER_SRC_OPTIMAL,
+				stagingImageID,
+				VkImageLayout_VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL,
+				1,
+				imageBlitData.Ptr(),
+				filter,
+			).AddRead(imageBlitData.Data()),
+			// Set the blit image to transfer src
+			cb.VkCmdPipelineBarrier(
+				commandBufferID,
+				VkPipelineStageFlags(VkPipelineStageFlagBits_VK_PIPELINE_STAGE_ALL_COMMANDS_BIT),
+				VkPipelineStageFlags(VkPipelineStageFlagBits_VK_PIPELINE_STAGE_ALL_COMMANDS_BIT),
+				VkDependencyFlags(0),
+				0,
+				memory.Nullptr,
+				0,
+				memory.Nullptr,
+				1,
+				stagingImageToSrcBarrierData.Ptr(),
+			).AddRead(
+				stagingImageToSrcBarrierData.Data(),
+			),
+		)
+	}
+
+	writeEach(ctx, out,
+		cb.VkCmdCopyImageToBuffer(
+			commandBufferID,
+			copySrc,
+			VkImageLayout_VK_IMAGE_LAYOUT_TRANSFER_SRC_OPTIMAL,
+			bufferID,
+			1,
+			bufferImageCopyData.Ptr(),
+		).AddRead(
+			bufferImageCopyData.Data(),
+		),
+	)
+
+	writeEach(ctx, out,
+		// Reset the image, and end the command buffer.
+		cb.VkCmdPipelineBarrier(
+			commandBufferID,
+			VkPipelineStageFlags(VkPipelineStageFlagBits_VK_PIPELINE_STAGE_ALL_COMMANDS_BIT),
+			VkPipelineStageFlags(VkPipelineStageFlagBits_VK_PIPELINE_STAGE_ALL_COMMANDS_BIT),
+			VkDependencyFlags(0),
+			0,
+			memory.Nullptr,
+			0,
+			memory.Nullptr,
+			1,
+			attachmentImageResetLayoutBarrierData.Ptr(),
+		).AddRead(
+			attachmentImageResetLayoutBarrierData.Data(),
+		),
+		cb.VkEndCommandBuffer(
+			commandBufferID,
+			VkResult_VK_SUCCESS,
+		),
+	)
+
+	// Submit all the commands above, wait until finish.
+	writeEach(ctx, out,
+		cb.VkDeviceWaitIdle(vkDevice, VkResult_VK_SUCCESS),
+		cb.VkQueueSubmit(
+			vkQueue,
+			1,
+			submitInfoData.Ptr(),
+			fenceID,
+			VkResult_VK_SUCCESS,
+		).AddRead(
+			submitInfoData.Data(),
+		).AddRead(
+			commandBuffers.Data(),
+		),
+		cb.VkWaitForFences(
+			vkDevice,
+			1,
+			fenceData.Ptr(),
+			1,
+			0xFFFFFFFFFFFFFFFF,
+			VkResult_VK_SUCCESS,
+		).AddRead(
+			fenceData.Data(),
+		),
+		cb.VkDeviceWaitIdle(vkDevice, VkResult_VK_SUCCESS),
```

Comment:
Do we need to this wait here? as already waited for the fence and we already used the baerries for the resovle/blit operation.

Reply:
The overdraw code was not changed from the original (This is what used to be in read_framebuffer that was shared). I didn't want to touch it too much, since it was working. There is some follow-up work to re-write the overdraw, and make it work more like the read_framebuffer does now.